File: templates/llm-output-r-eval-parse-detector/examples/bad.R
# Bad fixture: 5+ instances of the eval(parse(text=...)) anti-idiom in R.

# 1: classic dynamic-variable assignment
for (i in 1:5) {
  eval(parse(text = paste0("model_", i, " <- lm(y ~ x, data = df)")))
}

# 2: dynamic formula via sprintf
build_and_run <- function(col) {
  eval(parse(text = sprintf("summary(df$%s)", col)))
}

# 3: column-name interpolation from a function arg (injection sink)
get_value <- function(name) {
  eval(parse(text = paste0("df$", name)))
}

# 4: fully qualified base::eval — still the same anti-idiom
base::eval(parse(text = "x <- 1 + 1"))

# 5: evalq form
evalq(parse(text = "y <- 2"))

# 6: str2lang variant — also flagged
run_str <- function(s) {
  eval(str2lang(s))
}

# 7: str2expression variant
run_expr <- function(s) {
  eval(str2expression(s))
}

# 8: multi-line call (still detected)
eval(
  parse(
    text = paste0("z <- ", 1, " + ", 2)
  )
)

File: templates/llm-output-clojure-eval-read-string-detector/examples/bad.clj
;; Bad fixture: 8 instances of the (eval (read-string ...)) anti-idiom in Clojure.

;; 1: classic dynamic def via string concatenation
(doseq [i (range 5)]
  (eval (read-string (str "(def model-" i " (fit data))"))))

;; 2: dynamic form via str — column-name interpolation (injection sink)
(defn get-value [name]
  (eval (read-string (str "(:" name " row)"))))

;; 3: fully qualified clojure.core/eval — still the same anti-idiom
(clojure.core/eval (read-string "(+ 1 1)"))

;; 4: fully qualified clojure.core/read-string
(eval (clojure.core/read-string "(println :hi)"))

;; 5: load-string is the one-shot string-eval primitive
(load-string "(def x 42)")

;; 6: fully qualified load-string
(clojure.core/load-string "(def y 99)")

;; 7: multi-line spelling (still detected)
(eval
  (read-string
    (str "(def z " 1 " + " 2 ")")))

;; 8: load-string fed from a function arg (worst case)
(defn run-user-snippet [s]
  (load-string s))
